### PR TITLE
Fix for delDriveData replay issue

### PR DESCRIPTION
**Pull request by invisig0th**

### CODE REVIEW

File: synapse/lib/cell.py [hunk 1]
```diff
@@ -1854,8 +1854,8 @@ async def setDriveData(self, iden, versinfo, data):
         return self.drive.setItemData(iden, versinfo, data)
 
     @s_nexus.Pusher.onPushAuto('drive:data:del')
-    async def delDriveData(self, iden, vers=None):
-        return self.drive.delItemData(iden, vers=vers)
+    async def delDriveData(self, iden, vers):
+        return self.drive.delItemData(iden, vers)
```
Comment (vEpiphyte) [inline synapse/lib/cell.py:1858]: Is there still a reason for drive.delItemData / drive._delItemData to use a default vers=None ? It feels like if someone were to pass `vers=None` they would end up with a replay-unsafe nexus event.

Comment (invisig0th) [reply synapse/lib/cell.py:1858]: The implementation of the Drive class is not nexus aware or nexus safe. It's up to the Cell to implement nexus safe APIs.

Comment (invisig0th) [reply synapse/lib/cell.py:1858]: Actually, i like the "whatever is current" use case. I'll divide `delDriveData()` into a top/bottom half where the top half can lookup the current version if it's not specified.

Comment (vEpiphyte) [reply synapse/lib/cell.py:1864]: good call on splitting it up :)

File: synapse/lib/cell.py [hunk 2]
```diff
@@ -1853,9 +1853,15 @@ async def setDriveInfoPath(self, iden, path):
...
 
-    @s_nexus.Pusher.onPushAuto('drive:data:del')
     async def delDriveData(self, iden, vers=None):
-        return self.drive.delItemData(iden, vers=vers)
+        if vers is None:
+            info = self.drive.reqItemInfo(iden)
+            vers = info.get('version')
+        return await self._push('drive:data:del', iden, vers)
+
+    @s_nexus.Pusher.onPushAuto('drive:data:del')
```
Comment (Cisphyx) [inline synapse/lib/cell.py:1862]:
```suggestion
    @s_nexus.Pusher.onPush('drive:data:del')
```

**Review by Cisphyx — APPROVED** [review]

**Review by vEpiphyte — APPROVED** [review]

**Pull request merged**